### DISCUSSION TITLE
jinja2.exceptions.TemplateNotFound

### DISCUSSION THREAD

**Discussion by eikesf** [apache/airflow#24419] [Q&A] [answered]
In my dag i've created a task to run a bash command ("cd /home/eike/Projetos/backups && sh restore.sh") and in the terminal it's running fine, but when i try to run the task in airflow i get the error saying jinja2.exceptions.TemplateNotFound

Error
`jinja2.exceptions.TemplateNotFound: cd /home/eike/Projetos/backups && sh restore.sh
`
My task
` restaurar_bd = BashOperator(
        task_id = "restaurar_banco",
        bash_command = "cd /home/eike/Projetos/backups && sh restore.sh",
        dag = dag
    )`

Does anyone know how can i solve this error ? Does i need to create a template file or something like that ?

**Comment by josh-fell** [comment, marked as answer]
Hi @eikesf! You can to add a space to the end of the `bash_command` to get this to execute cleanly. What's happening is that `bash_command` is a template field which also [automatically considers file names ending in `.sh` or `.bash` as a templated file](https://github.com/apache/airflow/blob/6d69dd062f079a8fbf72563fd218017208bfe6c1/airflow/operators/bash.py#L127-L130) which you pass as a string. Since the `bash_command` arg being passed ends with ".sh", Airflow is assuming this is pointing to a Jinja-templated shell script at attempts to render it as such. Most users add a space or semicolon to the end of the command string to resolve this.

The [`BashOperator` API documentation](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/operators/bash/index.html#airflow.operators.bash.BashOperator) has a note about this as well.

**Reply by eikesf** [reply]
Thanks for the reply Josh, adding this space after the .sh solved that problem. Now i'm having some permission problem with docker
